sonar-scanner-engine/src/main/java/org/sonar/scanner/genericcoverage/GenericCoverageSensor.java

```diff
@@ -82,39 +82,27 @@ public GenericCoverageSensor(Settings settings) {
 
   }
 
-  /**
-   * Use an initializer to migrate old properties to the new one before Sensor phase so that
-   * Sensor will not be executed if there is no report (thanks to SensorDescriptor.requireProperty(REPORT_PATH_PROPERTY_KEY))
-   */
-  @Override
-  public void execute() {
-    Set<String> reportPaths = new LinkedHashSet<>();
-    reportPaths.addAll(Arrays.asList(settings.getStringArray(REPORT_PATHS_PROPERTY_KEY)));
-    loadDeprecated(reportPaths, OLD_REPORT_PATH_PROPERTY_KEY);
-    loadDeprecated(reportPaths, OLD_COVERAGE_REPORT_PATHS_PROPERTY_KEY);
-    loadDeprecated(reportPaths, OLD_IT_COVERAGE_REPORT_PATHS_PROPERTY_KEY);
-    loadDeprecated(reportPaths, OLD_OVERALL_COVERAGE_REPORT_PATHS_PROPERTY_KEY);
-    if (!reportPaths.isEmpty()) {
-      settings.setProperty(REPORT_PATHS_PROPERTY_KEY, reportPaths.stream().collect(Collectors.joining(",")));
-    }
-  }
-
   private void loadDeprecated(Set<String> reportPaths, String propertyKey) {
-    if (settings.hasKey(propertyKey)) {
+    if (config.hasKey(propertyKey)) {
       LOG.warn("Property '{}' is deprecated. Please use '{}' instead.", propertyKey, REPORT_PATHS_PROPERTY_KEY);
-      reportPaths.addAll(Arrays.asList(settings.getStringArray(propertyKey)));
+      reportPaths.addAll(Arrays.asList(config.getStringArray(propertyKey)));
     }
   }
 
   @Override
   public void describe(SensorDescriptor descriptor) {
     descriptor.name("Generic Coverage Report")
-      .requireProperty(REPORT_PATHS_PROPERTY_KEY);
+      .onlyWhenConfiguration(c -> asList(REPORT_PATHS_PROPERTY_KEY, OLD_REPORT_PATH_PROPERTY_KEY, OLD_COVERAGE_REPORT_PATHS_PROPERTY_KEY,
+        OLD_IT_COVERAGE_REPORT_PATHS_PROPERTY_KEY, OLD_OVERALL_COVERAGE_REPORT_PATHS_PROPERTY_KEY)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "OLD_IT_COVERAGE_REPORT_PATHS_PROPERTY_KEY"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "OLD_OVERALL_COVERAGE_REPORT_PATHS_PROPERTY_KEY"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)